src/shader_recompiler/ir/ir_emitter.cpp

```diff
@@ -1359,8 +1359,17 @@ U1 IREmitter::IGreaterThan(const U32& lhs, const U32& rhs, bool is_signed) {
     return Inst<U1>(is_signed ? Opcode::SGreaterThan : Opcode::UGreaterThan, lhs, rhs);
 }
 
-U1 IREmitter::INotEqual(const U32& lhs, const U32& rhs) {
-    return Inst<U1>(Opcode::INotEqual, lhs, rhs);
+U1 IREmitter::INotEqual(const U32U64& lhs, const U32U64& rhs) {
+    if (lhs.Type() != rhs.Type()) {
+        UNREACHABLE_MSG("Mismatching types {} and {}", lhs.Type(), rhs.Type());
+    }
+    switch (lhs.Type()) {
+    case Type::U32:
+    case Type::U64:
+        return Inst<U1>(Opcode::INotEqual, lhs, rhs);
```

Comment:
INotEqual is defined as having U32 inputs https://github.com/shadps4-emu/shadPS4/blob/be4c38bf1c4ebd6bf669176cc5123cbf9103bd01/src/shader_recompiler/ir/opcodes.inc#L307
I suspect if you compile in debug mode it will assert on mismatched types if passed U64

Reply:
I'm pretty sure it didn't assert there in debug 🤔